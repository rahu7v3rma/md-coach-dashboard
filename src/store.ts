import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
