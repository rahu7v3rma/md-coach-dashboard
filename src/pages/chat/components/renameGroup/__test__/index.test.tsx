import React from 'react';
import { Provider } from 'react-redux';
import * as router from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { act, create } from 'react-test-renderer';

import RenameGroupContainer from 'src/pages/chat/components/renameGroup';
import store from 'src/store';

const navigate = jest.fn();

describe('RenameGroupContainer', () => {
    beforeEach(() => {
        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
        jest.clearAllMocks();
    });

    it('RenameGroupContainer snapshot', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onClose={() => {}}
                        onRename={() => {}}
                    />
                </BrowserRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });

    it('DashboardLayout renders without errors', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onClose={() => {}}
                        onRename={() => {}}
                    />
                </BrowserRouter>
            </Provider>
        );
        expect(tree.root).toBeTruthy();
    });

    it('renders with a title and close button', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onClose={() => {}}
                        onRename={() => {}}
                    />
                </BrowserRouter>
            </Provider>
        );
        let titleText = tree.root.findByProps({
            id: 'title_text'
        }).props;
        expect(titleText.children).toBe('Edit group details');
        let closeIcon = tree.root.findByProps({
            id: 'close_icon'
        }).props;
        expect(closeIcon).toBeTruthy();
    });

    it('updates the input value when typing', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onClose={() => {}}
                        onRename={() => {}}
                    />
                </BrowserRouter>
            </Provider>
        );
        let groupInput = tree.root.findByProps({
            id: 'group_input'
        }).props;
        await act(() =>
            groupInput.onChange({ target: { value: 'group name' } })
        );
        let groupInputUpdated = tree.root.findByProps({
            id: 'group_input'
        }).props;
        expect(groupInputUpdated.value).toBe('group name');
    });

    it('calls the onRename callback with the new group name', async () => {
        const onRename = jest.fn();
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onClose={() => {}}
                        onRename={onRename}
                    />
                </BrowserRouter>
            </Provider>
        );
        let groupInput = tree.root.findByProps({
            id: 'group_input'
        }).props;
        await act(() =>
            groupInput.onChange({ target: { value: 'group name new' } })
        );
        let btnRename = tree.root.findByProps({
            id: 'btn_rename'
        }).props;
        await act(() => btnRename.onClick());
        expect(onRename).toHaveBeenCalledWith('group name new');
    });

    it('calls the onClose callback when the close button is clicked', async () => {
        const onClose = jest.fn();
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'name'}
                        onRename={() => {}}
                        onClose={onClose}
                    />
                </BrowserRouter>
            </Provider>
        );
        let btnCancel = tree.root.findByProps({
            id: 'btn_cancel'
        }).props;
        await act(() => btnCancel.onClick());
        expect(onClose).toHaveBeenCalled();
    });

    it('renders the component with the initial group name', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <RenameGroupContainer
                        name={'initial name'}
                        onRename={() => {}}
                        onClose={() => {}}
                    />
                </BrowserRouter>
            </Provider>
        );
        let groupInputUpdated = tree.root.findByProps({
            id: 'group_input'
        }).props;
        expect(groupInputUpdated.value).toBe('initial name');
    });
});
