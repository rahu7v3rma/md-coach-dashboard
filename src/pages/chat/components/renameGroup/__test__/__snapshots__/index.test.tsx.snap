// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenameGroupContainer RenameGroupContainer snapshot 1`] = `
<div
  className="sc-bgqQcB iAtTuv"
>
  <div
    className="sc-gTRrQi kQmhZJ"
  >
    <span
      className="sc-ewnqHT elDAAT"
      id="title_text"
    >
      Edit group details
    </span>
    <img
      className="sc-fFGjHI fIIilU"
      id="close_icon"
      onClick={[Function]}
      src={Object {}}
    />
  </div>
  <div
    className="sc-dicizt OtyZp"
  >
    <input
      className="sc-gxYJeL eMsdlH"
      id="group_input"
      onChange={[Function]}
      value="name"
    />
  </div>
  <div
    className="sc-kPTPQs bZMQqw"
  />
  <div
    className="sc-ispOId dfyVaL"
  >
    <button
      className="sc-kcuKUB dyCBKN"
      id="btn_cancel"
      onClick={[Function]}
    >
      Cancel
    </button>
    <button
      className="sc-dcrmVg jfTqEK"
      id="btn_rename"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
